src/gpl/src/nesterovBase.cpp

```diff
@@ -1922,14 +1922,17 @@ void NesterovBase::updateAreas()
                           / static_cast<float>(whiteSpaceArea_);
 
   if (totalFillerArea_ < 0) {
-    log_->error(GPL,
-                303,
-                "Use a higher -density or "
-                "re-floorplan with a larger core area.\n"
-                "Given target density: {:.2f}\n"
-                "Suggested target density: {:.2f}",
-                targetDensity_,
-                uniformTargetDensity_);
+    log_->warn(
+        GPL,
+        303,
+        "Not enough movable area, negative totalFillerArea_ in updateAreas(). "
+        "Given target density: {:.2f}. "
+        "targetDensity set with uniform density: {:.2f}",
+        targetDensity_,
+        uniformTargetDensity_);
+    targetDensity_ = uniformTargetDensity_;
+    movableArea_ = whiteSpaceArea_ * targetDensity_;
+    totalFillerArea_ = movableArea_ - nesterovInstsArea();
```

Comment:
What happens after updating the target density and recaulculating the areas? The whole GPL flow is restarted or it continues from the previous state.

Reply:
it doesn't restart, we just use the suggested density instead of throwing an error.

Reply:
Makes sense to me, since we at least have the warning telling the user that the target density was modified.

I'm just worried with cases where the new density is much larger/smaller than the original one. Wouldn't it be a problem, since the original goal of the user would be changed?
Perhaps we should have some margin to allow changing the density, but if the difference to the original is too large we just give up. (I'm don't know if this is really necessary, just an idea).

Reply:
There are other things orbiting this situation which I would like to discuss further, probably on a call. I reiterate that this is specific divergence detection (which I don't see happening often). I believe this change has some imperfections, but I believe it is more beneficial and fast solution for Peter. 

Considering the user desired density, we already do a lot of changes on the density outside of the user's control. Both in TCL and when the area changes.

Reply:
I would just add that from a user's perspective, it is completely fine and expected to increase the density when optimizing. Experienced users know not to expect the initial density to be the final density. In this case, it's just moving the resizer into GPL so the density increase happens in GPL rather than later on.

Honestly in commercial tools, this problem is worked around because the user doesn't set the target density. The user just provides the core area (or initial utilization to create the core area) and then the placer figures out the clustering density automatically somehow. It would be nice to more towards that at some point in the future.